lib/src/core/di/service_locator.dart

```diff
@@ -31,6 +34,30 @@ Future<void> initDependencies() async {
     ),
   );
 
+  // Register ChatRepository
+  serviceLocator.registerLazySingleton<ChatRepository>(
+    () => ChatRepositoryImpl(
+      // Pass required dependencies to the repository if needed
+    ),
+  );
+
+  // Register use cases
+  serviceLocator.registerLazySingleton<GetChatList>(
+    () => GetChatList(serviceLocator<ChatRepository>()),
+  );
+
+  serviceLocator.registerLazySingleton<GetArchivedList>(
+    () => GetArchivedList(serviceLocator<ChatRepository>()),
+  );
+
+  serviceLocator.registerLazySingleton<ArchiveChat>(
+    () => ArchiveChat(serviceLocator<ChatRepository>()),
+  );
+
+  serviceLocator.registerLazySingleton<DeleteChat>(
+    () => DeleteChat(serviceLocator<ChatRepository>()),
+  );
+
```

Comment:
Consider moving this setup into a chat_module.dart file inside the chat feature folder. This keeps service_locator.dart cleaner and respects feature boundaries.

Reply:
Missing registration of the use cases for injecting into the ChatBloc.